internal/querycoordv2/observers/handoff_observer.go

```diff
@@ -175,7 +175,7 @@ func (ob *HandoffObserver) schedule(ctx context.Context) {
 
 		case resp, ok := <-watchChan:
 			if !ok {
-				log.Error("watch Segment handoff loop failed because watch channel is closed!")
+				panic("watch Segment handoff loop failed because watch channel is closed!")
```

Comment:
should we panic here?

Reply:
Fix it!
usually when we lost connection with etcd, the service role will exit, panic here is meaningless.